sw/device/lib/dif/dif_warn_unused_result.h

```diff
@@ -11,6 +11,12 @@
  * This attribute must be used to mark all DIFs which return an error value of
  * some kind, to ensure that callers do not accidentally drop the error on the
  * ground.
+ *
+ * Normally, the correct way to drop such a value on the ground explicitly is
+ * with the syntax `(void)expr;`, in analogy with C++'s `[[nodiscard]]`
+ * attribute. Unfortunately, GCC incorrectly fails to implement this, so the
```

Comment:
I find that a rather harsh statement, given the long discussion in the bug you link to, and the fact that there's no agreement in there which behavior is more correct.

Reply:
Loosened up the language a bit. GCC is inconsistent in handling `[[nodiscard]]` and `warn_unused_result`, so we'll still note that in an ideal world, we would write `(void)x`.